### DISCUSSION TITLE
Upload Speed Limited to 10Mbps on Gigabit Connection with Nginx Proxy Manager Configuration

### DISCUSSION THREAD

**Discussion by psychedelicu** [nextcloud/all-in-one#4307] [Questions] [answered]
Hello,

Despite having a gigabit connection, my upload speed seems to be capped at 10Mbps. I've made several configurations to try and resolve the issue without success. My domain is on cloudflare. I bypassed cache, pause cloudflare, added page rules, nothing works!

NPM > Custom Nginx Configuration:

client_body_buffer_size 512k;
proxy_read_timeout 86400s;
client_max_body_size 0;
http3 off;
http3_hq off;

i get this error on Nginx Proxy manager logs:

2024/03/03 13:42:43 [warn] 5267#5267: *9000 a client request body is buffered to a temporary file /usr/local/nginx/client_body_temp/0000000714, client: 162.158.18.193, server: drive.domain.com, request: "PUT /remote.php/dav/uploads/psy/web-file-upload-f999ebda6b8dceae/3 HTTP/2.0", host: "drive.domain.com", referrer: "https://drive.domain.com/apps/files/files/1445?dir=/Test"

**Comment by psychedelicu** [comment]
also, how exactly do i edit the max_chunk_size in nextcloud aio @ unraid?

sudo -u www-data php occ config:app:set files max_chunk_size --value 20971520

**Reply by Zoey2936** [reply]
see https://github.com/nextcloud/all-in-one#how-to-run-occ-commands

**Reply by psychedelicu** [reply]
The only value which nextcloud is working is 10485760 (10mb). If i put --value 0 uploading is not working, and any other value produce errors in nextcloud related to files.

**Comment by psychedelicu** [comment]
I have removed http2 from my nginx proxy manager for nextcloud proxy by adding http2 off; to the config. Now the upload and download increased from 5-6mb to 10-15mb. But is still very low...

**Comment by szaimen** [comment]
Hi, do you have some warnings when you open `https://yourdomain.com/settings/admin/overview` ?

**Reply by psychedelicu** [reply]
Yes. 

![image](https://github.com/nextcloud/all-in-one/assets/35727400/5e783535-5b2d-4b60-8f8c-82423a88d551)

**Comment by szaimen** [comment, marked as answer]
See https://github.com/nextcloud/all-in-one/discussions/2475 then

**Reply by psychedelicu** [reply]
Still not working! I don't know why, but npmplus sometimes use http3/quic over http2, even if it's already disabled for the nextcloud domain.

On http3 speeds are very low, 50mbps.
On http2 speeds are 80-90mbps. 
With http2 disabled speeds are 140-160mbps, but the npmplus dev guy said i should not disable http2.

I have tried your solution, but is not working. The speeds are the same, even lower to 30mbps. 

![image](https://github.com/nextcloud/all-in-one/assets/35727400/6cbff28e-e79e-45d9-b6df-a89eb8f57438)
![image](https://github.com/nextcloud/all-in-one/assets/35727400/629a7737-9007-4e13-83e5-b55fefea3329)

Also, is full of errors. In the last 3-4 months, all I did is to try to get it working. 
I have tried on unraid, docker, ubuntu, debian, truenas, nextcloud aio, nextcloud vm, nextcloud server, nextcloud from casaos and cosmos os, with all of them i had tons of different issues., even with nextcloud android app. 

I need an honest opinion. Can this app really work? Or should I simply give up?

![image](https://github.com/nextcloud/all-in-one/assets/35727400/dc9981d5-059b-4a74-a616-af40d1747745)

**Reply by Zoey2936** [reply]
Can you maybe reenable http2 and http3 in your npmplus and send me a public downloadlink to a bigger file on your cloud, so I can check the download speeds I get with http2 / http3, please?
For http2, I don't recommend disabling it, since 1. using `http2 off;` breaks on restart as you've seen yourself, this could be fixed by changing one regex in a sed command in the start.sh of NPMplus (you can create a PR if you want), but 2. I think every website in 2024 should support http2. (also I'm a girl)

**Reply by psychedelicu** [reply]
Thank you! It seems that you are the only one willing to help me.
Everything is enabled:
![image](https://github.com/nextcloud/all-in-one/assets/35727400/78539dc9-a598-439c-bfb6-eb947a35d6ab)

Sure! Here it is: https://drive.withpsy.com/s/MmPK42eeqMWqHWx

I've just tested with seafile, and it seems that i have the same issue. Uploading is stuck at 90-98mbps. 

My CF records:
<img width="977" alt="image" src="https://github.com/nextcloud/all-in-one/assets/35727400/fcff2acf-9767-447a-a8ec-20b195b9bb2e">

**Reply by Zoey2936** [reply]
```
zoey@mx:~$ curl --http1.1 https://drive.withpsy.com/s/MmPK42eeqMWqHWx/download/MK.Ultra.2022.1080p.WEB-DL.DD5.1.H.264-CMRG.mkv -o l
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
 16 4910M   16  795M    0     0  62.6M      0  0:01:18  0:00:12  0:01:06 66.6M^C
zoey@mx:~$ curl --http2 https://drive.withpsy.com/s/MmPK42eeqMWqHWx/download/MK.Ultra.2022.1080p.WEB-DL.DD5.1.H.264-CMRG.mkv -o l
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
 29 4910M   29 1454M    0     0  37.3M      0  0:02:11  0:00:38  0:01:33 40.0M^C
zoey@mx:~$ curl --http3 https://drive.withpsy.com/s/MmPK42eeqMWqHWx/download/MK.Ultra.2022.1080p.WEB-DL.DD5.1.H.264-CMRG.mkv -o l
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
  0 4910M    0 16.9M    0     0  1244k      0  1:07:21  0:00:13  1:07:08 1388k^C
```
yes http3 seems to be very slow, but http2 is only a bit slower than http1.1 - but sorry I have no Idea what this causes, can you maybe try caddy and check the download speed with it?

**Reply by psychedelicu** [reply]
I've tried already... Caddy, Traefik, NPM.. Nothing works. I have an Omada ER605 tp link router. Maybe this is where should I look? What setting can cause this limit? Because I searched the internet and changed every possible setting...

**Reply by Zoey2936** [reply]
I think it needs to be your network, since with my NPMplus:
```
runner@1blu:~$ ./curl <url> -o l --http1.0
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100  109M  100  109M    0     0  63.7M      0  0:00:01  0:00:01 --:--:-- 63.8M
runner@1blu:~$ ./curl <url> -o l --http1.1
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100  109M  100  109M    0     0  67.1M      0  0:00:01  0:00:01 --:--:-- 67.1M
runner@1blu:~$ ./curl <url> -o l --http2
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100  109M  100  109M    0     0  63.5M      0  0:00:01  0:00:01 --:--:-- 63.5M
runner@1blu:~$ ./curl <url> -o l --http3
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100  109M  100  109M    0     0  11.9M      0  0:00:09  0:00:09 --:--:-- 12.3M
```

**Reply by psychedelicu** [reply]
I figured out! It was the damn router or omada software controller. I just reset them all, controller, router and switch. It was a pain to configure them again but I think it was worth it. Next time I will go with ubiquiti...

Maybe these settings caused this issue? 

![image](https://github.com/nextcloud/all-in-one/assets/35727400/7c882ac0-bdf3-4b5e-9750-4a277bdb4ca6)
![image](https://github.com/nextcloud/all-in-one/assets/35727400/d4aa68f2-eb85-48af-be68-b19a597b3740)
![image](https://github.com/nextcloud/all-in-one/assets/35727400/4ac510d4-b923-44ca-bda1-64a6d0b203ab)

**Reply by Zoey2936** [reply]
so http2 is now faster?

**Reply by psychedelicu** [reply]
yes! but i noticed when the server starts for the first time, it uses http3. I have to restart npmplus or save the domain proxy again. http3 is already disabled in the config, and i want it off, because http2 is way faster than 3. 

![image](https://github.com/nextcloud/all-in-one/assets/35727400/89ad5239-2697-49cc-887b-a17235911df7)

**Comment by psychedelicu** [comment]
I want to get at least half of this result...

![image](https://github.com/nextcloud/all-in-one/assets/35727400/d1f0c137-7f33-4be5-833c-d7bcbc77c5ff)
